run_operations.py

```diff
@@ -1,394 +1,351 @@
+            5. Runs `process_rsmas.py` if their is new data available
+        6. Waits for all of the output files from submitted `process_rsmas` calls to exist
+    :param args: command line arguments to use
+    """
+    inps = command_line_parse(args)
+
+    # Remove and reinitiate $OPERATIONS directory
+    if inps.restart:
+        shutil.rmtree(OPERATIONS_DIRECTORY)
+    
+    initiate_operations()
+
+    template_files = []
+
+    # inps.sheet_ids is an array of sheets.
+    # Each use of the --sheet_id command line parameter adds another array to the inps.sheet_id variable.
+    for sheet in inps.sheet_ids:
+        template_files += generate_templates_with_options(inps.template_csv, inps.dataset, sheet[0])
+
+    datasets = get_datasets_to_process(template_files, inps.dataset)
+
+    logger_run_operations.log(loglevel.INFO, "Datasets to Process: {}".format(datasets))
+
+    output_files = []
+    job_to_dset = {}
+
+    for dset in datasets:
+
+        template_file = "{}/{}.template".format(TEMPLATE_DIRECTORY, dset)
+
+        logger_run_operations.log(loglevel.INFO, "{}: {}".format(dset, template_file))
+
+        newest_date = get_newest_data_date(template_file)
+        last_date = get_last_downloaded_date(dset)
+
+        logger_run_operations.log(loglevel.INFO, "Newest Date: {}".format(newest_date))
+        logger_run_operations.log(loglevel.INFO, "Last Date: {}".format(last_date))
 
+        if newest_date > last_date:
+
+            logger_run_operations.log(loglevel.INFO, "Starting process_rsmas for {}".format(dset))
+
+            #  Exit and don't overwrite date file if process_rsmas.py throws and error
+            try:
+                # Submit processing job and running processing routine vis process_rsmas.py
+                outputs, job = run_process_rsmas(inps, template_file, dset)
+
+                # Overwrite the most recent date of data download in the date file
+                overwrite_stored_date(dset, newest_date)
+
+                job_to_dset[job] = dset
+                output_files += outputs
+
+            except Exception as e:
+                logger_run_operations.log(loglevel.ERROR, "process_rsmas threw an error ({}) and exited.".format(e))
+
+        logger_run_operations.log(loglevel.INFO, "{} process_rsmas calls completed.".format(dset))
+
+    logger_run_operations.log(loglevel.INFO, "All process_rsmas calls complete. Waiting for output files to appear")
+
+    logger_run_operations.log(loglevel.INFO, "{} total output files".format(len(output_files)))
+
+    # check if output files exist
+    i = 0
+    wait_time_sec = 60
+    total_wait_time_min = 0
+    while i < len(output_files):
+        if os.path.isfile(output_files[i]):
+            logger_run_operations.log(loglevel.INFO, "Job #{} of {} complete (output file {})".format(i + 1, len(output_files), output_files[i]))
+            dset = job_to_dset[output_files[i].split("process_rsmas_")[1].split(".")[0]]
```

Comment:
Can you provide a comment that gives an example of what this line does? Or can you provide a single line comment explanation?